File: loadercomplex-fabric/ver17/src/main/java/com/enderzombi102/loadercomplex/fabric17/impl/block/FabricBlockstate.java
package com.enderzombi102.loadercomplex.fabric17.impl.block;

import com.enderzombi102.loadercomplex.api.minecraft.block.Blockstate;
import com.enderzombi102.loadercomplex.api.minecraft.util.ResourceIdentifier;
import net.minecraft.block.BlockState;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class FabricBlockstate implements Blockstate {
	private final BlockState backingState;

	public FabricBlockstate(BlockState state) {
		backingState = state;
	}

	@Override
	public ResourceIdentifier getBlockType() {
		Identifier id = Registry.BLOCK.getId( backingState.getBlock() );
		return new ResourceIdentifier( id.getNamespace(), id.getPath() );
	}

	@Override
	public Object getObject() {
		return backingState;
	}
}

File: loadercomplex-fabric/src/ver171/kotlin/com/enderzombi102/loadercomplex/fabric171/impl/block/FabricBlockState.kt
package com.enderzombi102.loadercomplex.fabric171.impl.block

import com.enderzombi102.loadercomplex.api.minecraft.block.BlockState
import com.enderzombi102.loadercomplex.api.minecraft.util.ResourceIdentifier
import com.enderzombi102.loadercomplex.fabric171.impl.utils.toLC
import net.minecraft.util.registry.Registry
import net.minecraft.block.BlockState as McBlockState


class FabricBlockState( private val backingState: McBlockState ) : BlockState {
	override fun getBlockType(): ResourceIdentifier =
		Registry.BLOCK.getId( backingState.block ).toLC()

	override fun getObject(): Any =
		backingState
}
